ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -708,6 +754,59 @@ func init() {
 	klog.InitFlags(nil)
 }
 
+// addPod allows next goroutine to start once the webhook PodInformer cache updates
+func (t *TPUWebhookServer) addPod(obj interface{}) {
+	pod := obj.(*corev1.Pod)
+	klog.V(1).InfoS("addPod", "Pod", pod.Namespace+"/"+pod.Name, "Time", time.Now())
+
+	if t.lastAdmitted == "" {
+		// There is not a pending TPU worker Pod to the informer cache, unblock if waiting and return
+		for {
+			if t.waiting == 0 {
```

Comment:
this could be simplified to `for t.waiting > 0`

Reply:
Applied this suggestion in [8f4a349](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/859/commits/8f4a349cc4bc444010085ddcb5bef359c4f154dc).